core/storage/base_model/gae_models.py

```diff
@@ -295,6 +293,33 @@ def _reconstitute_from_snapshot_id(self, snapshot_id):
         reconstituted_model.last_updated = snapshot_model.last_updated
         return reconstituted_model
 
+    def _reconstitute_from_snapshot_id(self, snapshot_id):
+        """Makes this instance into a reconstitution of the given snapshot."""
+        snapshot_model = self.SNAPSHOT_CONTENT_CLASS.get(snapshot_id)
+        return self._reconstitute_from_snapshot_model(snapshot_model)
+
+    @classmethod
+    def _reconstitute_multi_from_models(cls, snapshot_id_model_list):
+        """Iterates through each entry in the list and reconstitutes the model
+        with the corresponding snapshot ID. Each value in the list is a tuple
+        where the first entry is a snapshot ID and the second is a model object.
+        This function should be more efficient than using
+        _reconstitute_from_snapshot_id for N models. Returns a list of the
+        reconstituted models.
+        """
+        snapshot_ids = [
+            snapshot_id_model[0]
+            for snapshot_id_model in snapshot_id_model_list]
+        models_to_reconstitute = [
+            snapshot_id_model[1]
+            for snapshot_id_model in snapshot_id_model_list]
+        snapshot_models = cls.SNAPSHOT_CONTENT_CLASS.get_multi(snapshot_ids)
+        return [
+            # pylint: disable=protected-access
```

Comment:
This should only apply to the given line -- put it on the right of the line below.